community/cypher/cypher/src/main/java/org/neo4j/cypher/export/SubGraphExporter.java

```diff
@@ -106,8 +107,21 @@ private void output( PrintWriter out, String ... commands )
 
     private Collection<String> exportConstraints()
     {
-        final List<String> result = new ArrayList<>();
-        for ( ConstraintDefinition constraint : graph.getConstraints() )
+        Iterable<ConstraintDefinition> constraints = graph.getConstraints();
+        int count = 0;
+        if ( constraints instanceof Collection )
+        {
+            count = ((Collection<ConstraintDefinition>) constraints).size();
+        }
+        else
+        {
+            for ( ConstraintDefinition ignored : constraints )
+            {
+                count++;
+            }
+        }
+        final List<String> result = new ArrayList<>( count );
```

Comment:
This is quite a bit more code in order to only be able to pre-size the collection. Is this method really hot enough to warrant it?

Reply:
I was asking myself the same question. SonarQube is referring to not pre-sized collections as a performance bug so I wanted to discuss this one